### PR TITLE
Fix price sorting

### LINKED ISSUE #61: Sorting by price is broken when using a comma separator
Sorting Gear by price, they are displayed with a comma. But this appears to break sorting order, where “100” sorts after “10,500”. Standard numeric sorting is also broken, where 25 sorts after 200.

I think you’re going to have to convert them from strings to integers, then convert them back on display.

Note that if the site is going to be localized into other languages, they might not use a comma as their thousands separator, so you might not want to bake that in.

This might also require breaking out the “(R)” for Restricted into a separate column.

### PR DESCRIPTION
**Pull request by AndreLemay**
Fixes #61 

Also adds a ColumnFactory that currently handles inserting the generatedId column that all tables need, and the hidden restricted column for tables with a price column. This can also be extended if we need to add additional columns solely for data in custom renderers or other functions that get the row data (such as #45)

### CODE REVIEW

**Comment by AndreLemay** [comment]
Not sure why it's including all the commits from my work on the datatable conversion, but the actual file changes seem correct to me.

**Comment by duffn** [comment]
It's no problem. 

And, again, thank you! Taking a look now.

**Comment by duffn** [comment]
Can you update these new armor records please? I must've entered these in there after you already updated that file. https://github.com/duffn/theouterrim/pull/77/files#diff-ebb6c0154ef8acee2b4fdefed2331836R674

**Comment by AndreLemay** [comment]
Updated

File: src/data/weapons.yaml [hunk 1]
```diff
@@ -917,7 +987,8 @@
   range: Extreme
   encumbrance: 10
   hp: 4
-  price: (R) 6,500
+  price: 6500
+  restricted: trie
```
Comment (duffn) [inline src/data/weapons.yaml]: `true`

**Comment by duffn** [comment]
This does look great and does fix sorting. Thank you.

Can you also please update the formatting of the price on the individual cards?

Screen Shot 2020-02-26 at 6 15 21 AM

File: src/components/shared/IndividualCard.js [hunk 2]
```diff
@@ -76,6 +76,14 @@ function renderField({ key, item, classes }) {
           <span className={classes.label}>Force Sensitive:</span> {item[key]}
         </Typography>
       )
+    case "price":
+      return (
+        <Typography key={key}>
+          <span className={classes.label}>Price:</span> {`${item.restricted ? "(R) " : ""}${item[key]}`}
```
Comment (duffn) [inline src/components/shared/IndividualCard.js]: I should have been clearer in my last comment, but can you also humanize this exactly like it is in the table? i.e. `(R) 1,400`

Comment (AndreLemay) [reply src/components/shared/IndividualCard.js]: Oooh yeah, my bad. Forgot to add the .toLocaleString() at the end.

**Pull request merged**